WcaOnRails/db/migrate/20160502113530_add_preferred_events_to_users.rb

```diff
@@ -0,0 +1,5 @@
+class AddPreferredEventsToUsers < ActiveRecord::Migration
+  def change
+    add_column :users, :preferred_event_ids, :text
```

Comment:
You're right, using the ids of the events as the ids of the `preferred_events` table doesn't make sense. This should work: `user.preferred_events = form_ids.map { |event_id| PreferredEvent.new(user_id: user, event_id: id) }` I'm not sure how this handles removing rows, though (e.g. if you update your preferred events and one event is missing this row should be removed as well).

About the `private` stuff: Skip it. I thought it would be a good idea to prevent people from setting non-sensical event ids, but ActiveRecord's API is too huge to hide all implementation details and this validation should go into `PreferredEvent` anyway.